### PR TITLE
feat: added functionality in 'add members in group' like 'add collaborators in project'

### LINKED ISSUE #2562: feat: Add functionality in "add members in group" like "add collaborators in project"
**Is your feature request related to a problem? Please describe.**
- Add functionality in "add members in group" like "add collaborators in project"
- add members in group
bandicam 2021-10-20 12-16-14-999

- add collaborators in project
bandicam 2021-10-20 12-15-54-826

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by vedant-jain03**
Fixes #2562 

#### Describe the changes you have made in this PR -
- added CDN link of bootstrap.tagsinput
- added the functionality similar to 'add collaborator'.

### Screenshots of the changes (If any) -
- before
bandicam 2021-10-20 12-16-14-999

- after
Screenshot (165)

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by vedant-jain03** [comment]
@pavanjoshi914 @tachyons CI test cases passed. Kindly Review

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -13,7 +14,7 @@
 @import 'footer';
 @import 'about';
 @import 'contribute';
-@import 'teacher';
+@import 'teacher';
```
Comment (tachyons) [inline app/assets/stylesheets/application.scss:17]: remove extra space here

Comment (vedant-jain03) [reply app/assets/stylesheets/application.scss]: @tachyons Done sir!

**Comment by pavanjoshi914** [comment]
Please make code climate happy!

File: app/javascript/controllers/addMember_controller.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        $(document).ready(() => {
+            $('#group_member_emails').tagsinput({
+                trimValue: true,
+                confirmKeys: [13, 44, 32],
+            });
+
+            $('.bootstrap-tagsinput input').attr('maxlength', '30');
+            $('.bootstrap-tagsinput input').attr('id', 'group_email_input');
+            $('.add-members-button').attr('disabled', true);
+            $('.bootstrap-tagsinput input').on('keyup', () => {
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/javascript/controllers/addMember_controller.js:17]: This block of code is for 'keyup' eventlistener.

File: app/javascript/controllers/addMember_controller.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                if ($('.bootstrap-tagsinput').children().length > 1) {
+                    $('.add-members-button').attr('disabled', false);
+                }
+                else {
+                    $('.add-members-button').attr('disabled', true);
+                }
+            });
+            $('.bootstrap-tagsinput').on('click', () => {
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/javascript/controllers/addMember_controller.js:25]: This block of code is for 'click' eventlistener.

**Comment by vedant-jain03** [comment]
@pavanjoshi914 Code Climate almost got happy.
@tachyons

**Comment by pavanjoshi914** [comment]
Is it possible to use jquery multiple class selectors in this case? to avoid duplication of code

**Comment by pavanjoshi914** [comment]
```package.json``` is already pushed by you for dependencies. no need of committing ```yarn.lock```

**Comment by vedant-jain03** [comment]
> Is it possible to use jquery multiple class selectors in this case? to avoid duplication of code

Actually Both functions are independent of each other. One is for 'click' event and second one is for 'keyup' event but the conditional loop is same for both of them.

**Comment by vedant-jain03** [comment]
> `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`

okhay I have removed package.json changes.

**Comment by pavanjoshi914** [comment]
> > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> 
> okhay I have removed package.json changes.

You have to remove yarn.lock. package.json is required

**Comment by vedant-jain03** [comment]
> > > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> > 
> > 
> > okhay I have removed package.json changes.
> 
> You have to remove yarn.lock that is what I said. package.json is required

Actually I have install both dependencies using yarn only that is why I have removed package.json.

**Comment by pavanjoshi914** [comment]
> > > > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> > > 
> > > 
> > > okhay I have removed package.json changes.
> > 
> > 
> > You have to remove yarn.lock that is what I said. package.json is required
> 
> Actually I think may be yarn.lock file is important as I have setup stimulus using webpacker and webpacker has installed yarn files that is important for the process. For tagsinput I have used yarn.

dependencies are preserved in ```package.json``` lock files are autogenerated

**Comment by vedant-jain03** [comment]
> > > > > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> > > > 
> > > > 
> > > > okhay I have removed package.json changes.
> > > 
> > > 
> > > You have to remove yarn.lock that is what I said. package.json is required
> > 
> > 
> > Actually I think may be yarn.lock file is important as I have setup stimulus using webpacker and webpacker has installed yarn files that is important for the process. For tagsinput I have used yarn.
> 
> dependencies are preserved in `package.json` lock files are autogenerated

I have removed `yarn.lock` file.

**Comment by pavanjoshi914** [comment]
> > Is it possible to use jquery multiple class selectors in this case? to avoid duplication of code
> 
> Actually Both functions are independent of each other. One is for 'click' event and second one is for 'keyup' event but the conditional loop is same for both of them.

Yes, we can trigger multiple on the same function, can you try this 

https://stackoverflow.com/questions/2534089/jquery-multiple-events-to-trigger-the-same-function

**Comment by tachyons** [comment]
> dependencies are preserved in package.json lock files are autogenerated

True, but not all packages follow proper versioning, So it is important to commit yarn.lock when pushing code

File: app/javascript/controllers/addMember_controller.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { Controller } from 'stimulus';
+import 'bootstrap-tagsinput/dist/bootstrap-tagsinput.js';
+import * as $ from 'jquery/dist/jquery';
+import 'bootstrap/dist/js/bootstrap';
+
+export default class extends Controller {
+    connect() {
```
Comment (tachyons) [inline app/javascript/controllers/addMember_controller.js:7]: Can you consider refactoring using  data-action https://stimulus.hotwired.dev/handbook/building-something-real

**Comment by pavanjoshi914** [comment]
> > dependencies are preserved in package.json lock files are autogenerated
> 
> True, but not all packages follow proper versioning, So it is important to commit yarn.lock when pushing code

@vedant-jain03  can you commit ```yarn.lock``` again

File: app/javascript/controllers/addMember_controller.js (hunk 4 above)
Comment (vedant-jain03) [reply app/javascript/controllers/addMember_controller.js:7]: Sure Sir.

**Comment by vedant-jain03** [comment]
@tachyons @pavanjoshi914 Kindly Review
1. added yarn.lock
2. Refactor the code using data-action (click) for addMember button in show.html.erb .

**Review by tachyons — APPROVED** [review]

**Comment by pavanjoshi914** [comment]
> @tachyons @pavanjoshi914 Kindly Review
> 
> 1. added yarn.lock
> 2. Refactor the code using data-action (click) for addMember button in show.html.erb .

Great work @vedant-jain03 !!  There are lot of commits out there can you make commit history a little clean with proper commit message. 

I suggest following commitlint convention as a good practice

**Comment by vedant-jain03** [comment]
> > @tachyons @pavanjoshi914 Kindly Review
> > 
> > 1. added yarn.lock
> > 2. Refactor the code using data-action (click) for addMember button in show.html.erb .
> 
> Great work @vedant-jain03 !! There are lot of commits out there can you make commit history a little clean with proper commit message.
> 
> I suggest following commitlint convention as a good practice

Thank you @pavanjoshi914 for the recommendation, I have edited my last commit message, actually stimulus was new for me and I have faced rspec testcases for the first time that is why it took me many commits. But learned a lot from every commit. 
I will try to have clean commit history from next time.

**Review by nitin10s — APPROVED** [review]
@tachyons

**Pull request merged**